x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -164,9 +193,53 @@ class KeyedTimeslider extends Component<Props, State> {
     });
   }
 
+  _updateTimeRange = (data: { label: string; ms: number; default?: boolean }) => {
+    if (!data.default) {
```

Comment:
Why is there a special case for default? Instead of passing back {ms: 0, default: true}, How about just passing back { ms: this.defaultRange.step } so a special case is not needed to handle default. You can pass autoStepSize into TimeSliderPopover and then just return autoStepSize in the callback when "auto" is selected instead of all this customized logic for the auto case.